_projects/2024/100535241/100535241.Rmd

```diff
@@ -0,0 +1,824 @@
+        panel.grid = element_line(color = "grey",
+                                  size = 0.2,
+                                  linetype = 1),
+        #axis.title.x = element_text(size = 16, family = "roboto", margin = margin(t = 10)),
+        axis.ticks.length.y = unit(0, "pt")) # to remove the axis gaps on the Male y-axis
+
+
+combined_plot
+
+```
+
+This is getting closer to a nice replication of the original. There is one major piece missing however, the legend.
+
+# Create the legend for the plot
+
+One of the biggest challenges for the replication has been the legend. The legend uses concentric circles that are outside of the plot area, have different colours to the actual plot and use specific call out labels.
+
+Because of these issues, I have built the legend manually then used `inset_element` to position it with the other plots. The legend is created as a grob, saved as an image and then called back and sized and positioned. Sizing and positioning within the two other plots has been a big challenge.
+
+```{r develop-legend, size='30%'}
+
+# Create data for concentric circles
+circle_data <- function(radius, npoints=100) {
+  angles <- seq(0, 2*pi, length.out = npoints)
+  data.frame(
+    x = radius * cos(angles),
+    y = radius * sin(angles)
+  )
+}
+
+# Create concentric circles data
+circlesmall <- circle_data(0.7)
+circlebig <- circle_data(3)
+
+# Plot concentric circles with grey fill
+concentric_plot <- ggplot() +
+  geom_polygon(data = circlebig, aes(x = x, y = y), fill = "grey90", color = "black") +
+  geom_polygon(data = circlesmall, aes(x = x, y = y - 2.3), fill = "grey90", color = "black")+
+  lims(x=c(-3,20))+
+  
+  annotate("segment", # line for <1k label
+           x=0, xend = 6,
+           y=-2.3, yend = -2.3,
+           size=1.5)+
+  annotate("segment", 
+           x = 0, xend = 6, 
+         y = 0, yend = 0,
+         size=1.5)+
+  annotate("text", x = 6.5, y = -2.3, label = "<1k individuals", size = 14, hjust = 0)+
+  annotate("text", x = 6.5, y = 0, label = "290k individuals", size = 14, hjust = 0)+
+  coord_fixed()+
+  theme_void()
+
+concentric_plot
+```
+
+We now save the plot created above as a .png to preserve our sizing ratios. If we overlay the raw grob, it distorts when joined to the plots.
+
+```{r save-legendpng, message = FALSE}
+png(file="~/Documents/UC3M - coursework/Data Vizualisation/dataviz/legend_test1.png",
+width=600, height=250)
+concentric_plot
+dev.off()
+
+```
+
+Now we read in the image and adjust and paste on our graph with inset_element.
+
+```{r read-image}
+img_path <- "~/Documents/UC3M - coursework/Data Vizualisation/dataviz/legend_test1.png"
+img <- magick::image_read(img_path)
+# Trim the image to remove unnecessary whitespace 
+img_trimmed <- magick::image_trim(img)
+# Make the white background transparent 
+img_transparent <- magick::image_transparent(img_trimmed, "white")
+
+img_grob <- rasterGrob(as.raster(img_transparent), interpolate = TRUE)
```

Comment:
Same idea as before: include the image in the PR and avoid paths that belong to your computer. :)